cranelift/codegen/src/machinst/vcode.rs

```diff
@@ -517,6 +527,17 @@ impl<I: VCodeInst> VCode<I> {
                 cur_srcloc = None;
             }
 
+            // Emit the constants used by the function.
+            let used_constants = buffer
+                .used_constants()
+                .map(|(c, l)| (*c, *l))
+                .collect::<Vec<_>>();
+            for (used_constant, label) in used_constants {
```

Comment:
This is actually inside the per-block loop body, so this would emit one copy of each (function-wide) constant per basic block. I don't think this is what was intended :-)